data/transactions/logic/evalAppTxn_test.go

```diff
@@ -2907,3 +2907,31 @@ itxn_submit
 
 	TestApp(t, source, ep, "appl depth (8) exceeded")
 }
+
+func TestForeignAppAccountAccess(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	ep, tx, ledger := MakeSampleEnv()
+	ledger.NewAccount(appAddr(888), 50_000)
+	tx.ForeignApps = []basics.AppIndex{basics.AppIndex(2)}
+
+	// This app looks itself up in the ledger, so we need to put it in there.
+	ledger.NewApp(tx.Sender, 2, basics.AppParams{
```

Comment:
Could you change this `2` to a higher number?  I'm just being a little paranoid because I want it to be obvious that the 2 is the appID of the app, and not a slot number in ForeignApps.

Also I think that comment is a cut-n-paste.  The app isn't looking itself up. The app being tested looks up this app.